test/hotspot/jtreg/compiler/vectorapi/TestVectorShiftImm.java

```diff
@@ -0,0 +1,450 @@
+    private static int[]   integersA, integersB;
+    private static long[]  longsA,    longsB;
+
+    private static byte  tBytes[][],    gBytes[][];
+    private static short tShorts[][],   gShorts[][];
+    private static int   tIntegers[][], gIntegers[][];
+    private static long  tLongs[][],    gLongs[][];
+
+    private static Random r = new Random(32781);
+
+    static final VectorSpecies<Byte> byte64SPECIES  = ByteVector.SPECIES_64;
+    static final VectorSpecies<Byte> byte128SPECIES = ByteVector.SPECIES_128;
+
+    static final VectorSpecies<Short> short64SPECIES  = ShortVector.SPECIES_64;
+    static final VectorSpecies<Short> short128SPECIES = ShortVector.SPECIES_128;
+
+    static final VectorSpecies<Integer> integer64SPECIES  = IntVector.SPECIES_64;
+    static final VectorSpecies<Integer> integer128SPECIES = IntVector.SPECIES_128;
+
+    static final VectorSpecies<Long> long128SPECIES = LongVector.SPECIES_128;
+
+    static String[] opNames = {"LSHL", "ASHR", "LSHR", "ASHR_AND_ACCUMULATE", "LSHR_AND_ACCUMULATE"};
+
+    static boolean allTestsPassed = true;
+    static StringBuilder errMsg = new StringBuilder();
+
+    public static void main(String args[]) {
+
+        int vlen = Integer.parseInt(System.getProperty("vlen", ""));
+
+        test_init();
+
+        if (vlen == 64) {
+            test_vlen64();
+        }
+
+        if(vlen == 128) {
+            test_vlen128();
+        }
+
+        if (allTestsPassed) {
+            System.out.println("Test PASSED");
+        } else {
+            throw new RuntimeException("Test Failed, failed tests:\n" + errMsg.toString());
+        }
+    }
+
+    static void test_vlen64() {
+        for (int i = 0; i < NUM_ITERS; i++) {
+            shift_and_accumulate(tBytes, true, byte64SPECIES, 64);
+            shift_and_accumulate(tShorts, true, short64SPECIES, 64);
+            shift_and_accumulate(tIntegers, true, integer64SPECIES, 64);
+
+            shift(tBytes, true, byte64SPECIES, 64);
+            shift(tShorts, true, short64SPECIES, 64);
+            shift(tIntegers, true, integer64SPECIES, 64);
+        }
+    }
+
+    static void test_vlen128() {
+        for (int i = 0; i < NUM_ITERS; i++) {
+            shift_and_accumulate(tBytes, true, byte128SPECIES, 128);
+            shift_and_accumulate(tShorts, true, short128SPECIES, 128);
+            shift_and_accumulate(tIntegers, true, integer128SPECIES, 128);
+            shift_and_accumulate(tLongs, true, long128SPECIES, 128);
+
+            shift(tBytes, true, byte128SPECIES, 128);
+            shift(tShorts, true, short128SPECIES, 128);
+            shift(tIntegers, true, integer128SPECIES, 128);
+            shift(tLongs, true, long128SPECIES, 128);
+        }
+    }
+
+    /**
+     * Tests for type byte.
+     */
+
+    static int shift_with_op(VectorOperators.Binary op, ByteVector vbb,
+                             byte arr[][], int end, int ind) {
+        vbb.lanewise(op, 1).intoArray(arr[end++], ind);
```

Comment:
How about adding case 0 in this test? Those code was changed.

Reply:
Thank you for watching this.
It can be covered by `lanewise(op, 8)`, `lanewise(op, 16)` and `lanewise(op, 24)`.
But I still added it, it looks clearer.

@theRealAph could you please take a look at the newest version?
The tests work well on our two very different aarch64 platforms.
Or the tests still looks too complicated?
Please let me know if you have any suggestions. Thanks.